### PR TITLE
SALTO-5215 - Salesforce: Improve dependency error text by updating the group names

### PR DESCRIPTION
**Pull request by tomsellek**
Change group names appear in user-visible error messages, so let's make them easier to understand by the end user.

---

TODO:
 - [ ] Final text from @tomermevorach 

---
_Release Notes_: 
Salesforce: Errors and warnings that contain the names of change groups are now clearer.

---
_User Notifications_: 
N/A

### CODE REVIEW

File: packages/salesforce-adapter/test/group_changes.test.ts [hunk 1]
```diff
@@ -151,12 +151,12 @@ describe('Group changes function', () => {
...
           .toEqual(changeGroupIds.get(anotherAddInstance.elemID.getFullName()))
         expect(changeGroupIds.get(addInstance.elemID.getFullName()))
-          .toEqual(`add_${customObjectName}_instances`)
+          .toEqual(`Add data instances of type '${customObjectName}'`)
       })
 
       it('should have a separate group for diff type', () => {
         expect(changeGroupIds.get(differentAddInstance.elemID.getFullName()))
-          .toEqual(`add_${differentCustomObjectName}_instances`)
+          .toEqual(`Add data instances of type '${differentCustomObjectName}'`)
```
Comment (amitgoldarb) [inline packages/salesforce-adapter/test/group_changes.test.ts:159]: If we add this group name to the dependency error in core/deploy/deploy_actions.ts:174, we receive a sentence that isn't complete.
I think this should be:
`addition/modifcation/removal of data instance of type`
cc @tomermevorach WDYT? as well

**Review by amitgoldarb — APPROVED** [review]
Approving to save cycles, I think we should edit the text a bit though cc @tomermevorach

File: packages/salesforce-adapter/test/group_changes.test.ts (hunk 1 above)
Comment (amitgoldarb) [reply packages/salesforce-adapter/test/group_changes.test.ts:159]: The text from the error in core is:
Element was not deployed, as it depends on ${groupName} which failed to deploy

Comment (tomermevorach) [reply packages/salesforce-adapter/test/group_changes.test.ts:159]: I am fine with @amitgoldarb's suggestion

**Pull request merged**